### PR TITLE
Fix experimentation updating

### PR DESCRIPTION
**Pull request by liyiy**
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

- fix updating experiments bug 
- fix copyto clipboard
- when update fails, don't exit edit mode

## How did you test this code?

Unit tests!

### CODE REVIEW

File: ee/clickhouse/views/experiments.py [hunk 1]
```diff
@@ -47,6 +47,9 @@ class Meta:
         ]
 
     def validate_feature_flag_key(self, value):
+        if self.context['request'].method == 'PUT' or self.context['request'].method == 'PATCH' and self.initial_data["feature_flag_key"] == value:
```
Comment (liyiy) [inline ee/clickhouse/views/experiments.py:50]: maybe there's a better way to check for this? 😕 @neilkakkar

Comment (neilkakkar) [reply ee/clickhouse/views/experiments.py:50]: what's the issue? O.o

Comment (neilkakkar) [reply ee/clickhouse/views/experiments.py:50]: ah, I think I see it: instead of patch, we send `create` request when trying to save an edited experiment?

Comment (neilkakkar) [reply ee/clickhouse/views/experiments.py:50]: Actually, after the serializer update I made in https://github.com/PostHog/posthog/pull/7981, we don't really need this validation here at all.

File: ee/clickhouse/views/experiments.py [hunk 2]
```diff
@@ -108,17 +111,7 @@ def create(self, validated_data: dict, *args: Any, **kwargs: Any) -> Experiment:
         return experiment
 
     def update(self, instance: Experiment, validated_data: dict, *args: Any, **kwargs: Any) -> Experiment:
-
-        expected_keys = set(["name", "description", "start_date", "end_date"])
```
[the remark this answers is not in the document]
Comment (neilkakkar) [reply ee/clickhouse/views/experiments.py:112]: Reason is that we've already created the FF by this point, so its annoying to try and go back and change that (although, we can do this if users do ask for it later on 🤷 )

[the remark this answers is not in the document]
Comment (liyiy) [reply ee/clickhouse/views/experiments.py:112]: I ended up disabling feature flag key editing and also check for it on the backend just to be sure 👍

File: ee/clickhouse/views/test/test_clickhouse_experiments.py [hunk 3]
```diff
@@ -232,17 +232,6 @@ def test_create_multivariate_experiment(self):
         id = response.json()["id"]
         end_date = "2021-12-10T00:00"
 
-        # Now try updating FF
-        response = self.client.patch(
-            f"/api/projects/{self.team.id}/experiments/{id}",
-            {
-                "description": "Bazinga",
-                "parameters": {"feature_flag_variants": [{"key": "control", "name": "X", "rollout_percentage": 100}]},
```
Comment (liyiy) [inline ee/clickhouse/views/test/test_clickhouse_experiments.py:240]: Are we allowing or not allowing users to be able to edit the "test groups"? @neilkakkar

Comment (neilkakkar) [reply ee/clickhouse/views/test/test_clickhouse_experiments.py:240]: Not allowing for now! (i.e. can't change variants / ff key)

But if you feel its easier, I don't mind deleting / updating existing FF with any new information that comes in^^

Comment (neilkakkar) [reply ee/clickhouse/views/test/test_clickhouse_experiments.py:240]: Hmm, I think you can focus on the other PR, I'll take this one over! Tried updating an experiment on cloud, indeed a pain, will allow updating groups and remove existing FF info!

**Pull request merged**